pennylane_lightning/src/bindings/Bindings.hpp

```diff
@@ -89,12 +93,90 @@ auto createManaged(
     }
     auto *data_ptr =
         static_cast<std::complex<PrecisionT> *>(numpyArrayInfo.ptr);
-    return StateVectorManaged<PrecisionT>(
-        {data_ptr, static_cast<size_t>(numpyArrayInfo.size)});
+    return StateVectorManagedCPU<PrecisionT>(
+        {data_ptr, static_cast<size_t>(numpyArrayInfo.shape[0])});
+}
+
+/**
+ * @brief Create numpy array view for the underlying data of
+ * `%StateVectorManagedCPU` object.
+ *
+ * @tparam PrecisionT Floating point data type
+ * @param sv `%StateVectorManagedCPU` object
+ * @return A numpy array
+ */
+template <class PrecisionT = double>
+auto toNumpyArray(const StateVectorManagedCPU<PrecisionT> &sv)
+    -> pybind11::array_t<std::complex<PrecisionT>> {
+    return pybind11::array_t<std::complex<PrecisionT>>(
+        {sv.getLength()}, {2 * sizeof(PrecisionT)}, sv.getData());
+}
+
+/**
+ * @brief Get memory alignment of a given numpy array.
+ *
+ * @param numpyArray Pybind11's numpy array type.
+ * @return Memory model describing alignment
+ */
+auto getNumpyArrayAlignment(const pybind11::array &numpyArray)
+    -> CPUMemoryModel {
+    return getMemoryModel(numpyArray.request().ptr);
+}
+
+/**
+ * @brief Create an aligned numpy array for a given type, memory model and array
+ * size.
+ *
+ * @tparam T Datatype of numpy array to create
+ * @param memory_model Memory model to use
+ * @param size Size of the array to create
+ * @return Numpy array
+ */
+template <typename T>
+auto alignedNumpyArray(CPUMemoryModel memory_model, size_t size)
```

Comment:
Just for my own understanding, numpy will maintain ownership of the data returned from this array?
Would it be possible to use a unique_ptr here, or does that break the memory model?

Reply:
Yes, one can assign a custom function as a destructor of a NumPy array. A problem with `std::unique_ptr` is that NumPy does not know how to treat it as an array.